hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/AssignmentManager.java

```diff
@@ -1176,15 +1176,9 @@ protected void periodicExecute(final MasterProcedureEnv env) {
         if (rsn.getState() != State.OPEN) {
           continue; // Opportunistic check, should quickly skip RITs, offline tables, etc.
         }
-        ServerName sn;
-        State state;
-        rsn.lock();
-        try {
-          sn = rsn.getRegionLocation();
-          state = rsn.getState();
-        } finally {
-          rsn.unlock();
-        }
+        // Should not acquire region stat lock, this may block other important work, see HBASE-23613.
+        ServerName sn = rsn.getRegionLocation();
+        State state = rsn.getState();
```

Comment:
This is metrics. The state and servername in RSN are both volatile. I think this change is good.

Reply:
"Do not need to acquire region state lock as this is only for showing metrics". This would be better?

Reply:
@Apache9 OK, change to it.